source/process/regex-library-draft.md

```diff
@@ -0,0 +1,175 @@
+* **unique footnote reference id**: use filename to make footnote reference id unique<br>F: `<sup class="fn" id="note-backlink-(\d+)"><a epub:type="noteref" href="([^#]+)_([^#]*?).xhtml#note-(\d+)">\[(\d+)\]</a></sup>`<br>R: `<sup class="fn" id="note-backlink-\3-\1"><a epub:type="noteref" href="\2_\3.xhtml#note-\3-\4">[\5]</a></sup>`
+* **unique footnote indicator id**: use filename to make footnote id unique<br>F: `<div id="note-(\d+)" epub:type="footnote">\s*<p><sup><a href="([^#]+)_([^#]*?)\.xhtml#note-backlink-(\d+)">`<br>R: `<div id="note-\3-\1" epub:type="footnote"><p><sup><a href="\2_\3.xhtml#note-backlink-\3-\4">`
+* **remove Ibids: **make sure footnotes are formatted correctly according to the style guide and then use to replace Ibids<br>F: `(<p class="[^"]*"><sup>(\d+)</sup>(.*?<span class="i">.*?</span>).*?</p>\s*<p class="[^"]*"><sup>\d+</sup>)Ibid\.(,.*?)*</p>`<br>R: `\1\3\4</p>`
+
+</details>
+
+<details close>
+
+<summary>Index</summary>
+
+* **move pagebreaks up top**: find pagebreaks in a file and move them before the h1. (<mark>Run multiple times until there are no new finds</mark>)<br>F: `(<h1[^>]*>.*?</h1>(?msi)(.*?))(<span epub:type="pagebreak"[^>]*></span>)`<br>R: `\3\1`
+
+</details>
+
+<details close>
+
+<summary>Links</summary>
+
+* **add `target="_blank"` to links**: Add `target="_blank"` attribute to existing external links<br>F: `<a href="http([^"]+)"><br>R: <a href="http\1" target="_blank" rel="noopener">`<br>R: `<a href="http\1" target="_blank" rel="noopener">`
+* **URLs**: Add links to URLs (Does not capture every instance)<br>F: `\shttp(.+?)([;|\.|,|\)][\s|<])`<br>R: `\s<a href="http\1" target="_blank" rel="noopener">http\1</a>\2\3`
+* **tag hyperlinks:** find and replace to tag hyperlinks<br>F: `<a (?:class="[^"]*"\s*)*href="((?:mail[^"]*)|(?:http[^"]*))">([^<]*)</a>`<br>R: `<a href="\1" target="_blank" rel="noopener">\2</a>`
+* **link chapters**: Find potential instances where chapters can be linked. Adjust the word `first` to `second` and the number `1` to `2` etc., to find all chapters<br>F: `(first chap(\.|ters?)|chap(s?\.|ters?) 1)(?!\d)`
+* **link parts**: Find potential instances where parts can be linked. Adjust the word `first` to `second` and the number `1` to `2` etc., to find all parts<br>F: `(first part|parts? 1)(?!\d)`
+
+</details>
+
+<details close>
+
+<summary>Percival</summary>
+
+* **percival parsing**: add parsing tags before headings containing scripture. Replace `Gen` with Bible book needed<br>F: `^(\s+)<(h\d)>(.*?)(\d+):(.*?)</\2>`<br>R: `\1<span data-parsing="Gen.\4"></span>\n\1<\2>\3\4:\5</\2>`
+
+</details>
+
+<details close>
+
+<summary>Commentary Markup</summary>
+
+* **headings `data-context`**: add `data-context` tags before headings. Adjust `h3` to capture desired heading<br>F: `^(\s+)<(h3)>(.*?<a data-ref="(.*?)">.*?</a>.*?)</\2>`<br>R: `\1<hr data-context="\4" />\n\1<\2>\3</\2>`
+
+</details>
+
+<details close>
+
+<summary>LESSON\_maker\_ Markup</summary>
```

Comment:
I'd like to remove this LESSONmaker section, since we won't be marking panic questions based solely on the criteria of their having `data-ref` s.